elrond-wasm/src/contract_base/wrappers/crypto_wrapper.rs

```diff
@@ -70,6 +71,17 @@ where
         A::crypto_api_impl().verify_ed25519(&key_bytes[..], message_byte_slice, &sig_bytes[..])
     }
 
+    pub fn sha256_managed<const SHA256_HASH_DATA_BUFFER_LEN: usize>(
+        &self,
+        attributes: &ManagedBuffer<A>,
+    ) -> ManagedByteArray<A, SHA256_RESULT_LEN> {
+        let attr_len = attributes.len();
+        let mut attributes_buffer = [0u8; SHA256_HASH_DATA_BUFFER_LEN];
+        let attributes_buffer_slice = &mut attributes_buffer[..attr_len];
+
```

Comment:
You also need to load the buffer onto the slice, as it is now the input is always zeroes.